benchmarks_test.go

```diff
@@ -197,6 +191,19 @@ func BenchmarkDB_SequentialInsert(b *testing.B) {
 	b.ReportMetric(float64(numObjectsToInsert*b.N)/b.Elapsed().Seconds(), "objects/sec")
 }
 
+func BenchmarkDB_Baseline_Part_Insert(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		tree := part.New[uint64](part.RootOnlyWatch)
+		txn := tree.Txn()
+		for j := uint64(0); j < numObjectsToInsert; j++ {
+			txn.Insert(index.Uint64(j), j)
+		}
+		tree = txn.Commit()
+		//require.Equal(b, tree.Len(), numObjectsToInsert)
```

Comment:
Do we need to replace with with:
```
if tree.Len() != numObjectsToInsert {
  t.Fatal()
}
```
?

I recall require.Equal being slow in benchmarks, but here its not replaced.

Reply:
I'll do a pass to make sure I don't call any `require/assert` in the loops. It's incredibly slow due to it calling `runtime.Callers`. That was also the reason why I called the `callerPackage` thing from `WriteTxn` as it was taking 30% of the time for small WriteTxn's.